### PR TITLE
Add producer to the hydrator handler using components lib

### PR DESCRIPTION
**Pull request by LorenBoren**
*Add usage of component-producer lib for producing hydrated assets
*Add usage of component-httpclient lib for a default Nexpose httpclient

### CODE REVIEW

File: pkg/hydrator/nexposeconfig.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+	httpclient "github.com/asecurityteam/component-httpclient"
+)
+
+// NexposeConfig holds configuration to connect to Nexpose
+// and make a call to the fetch assets API
+type NexposeConfig struct {
+	HTTPClient *httpclient.Config `description:"The HTTP client config from github.com/asecurityteam/component-httpclient."`
+	Host       string             `description:"The scheme and host of a Nexpose instance."`
+	Username   string             `description:"The username used to login to the Nexpose instance at the given host."`
```
Comment (bkrebsbach) [inline pkg/hydrator/nexposeconfig.go]: Suggestion: you leverage the `basicauth` plugin in transportd to remove these values from the code.

File: docker-compose.yaml [hunk 2]
```diff
@@ -5,29 +5,25 @@ services:
...
+      SERVERFULL_RUNTIME_CONNSTATE_ACTIVEGAUGE: http.server.connstate.active.gauge
+      SERVERFULL_RUNTIME_CONNSTATE_ACTIVECOUNTER: http.server.connstate.active
+      SERVERFULL_RUNTIME_CONNSTATE_NEWGAUGE: http.server.connstate.new.gauge
+      SERVERFULL_RUNTIME_CONNSTATE_NEWCOUNTER: http.server.connstate.new
+      SERVERFULL_RUNTIME_LOGGER_OUTPUT: STDOUT
+      SERVERFULL_RUNTIME_LOGGER_LEVEL: INFO
+      SERVERFULL_RUNTIME_STATS_OUTPUT: NULL
+      SERVERFULL_RUNTIME_SIGNALS_INSTALLED: OS
+      SERVERFULL_RUNTIME_SIGNALS_OS_SIGNALS: 15 2
+      NEXPOSE_HOST: ""
```
Comment (bkrebsbach) [inline docker-compose.yaml]: Task: these variable names are inaccurate. It'd be something like `VULNHYDRATOR_NEXPOSE_HOST`. I'd highly recommend documenting all the environment variables required in the README

File: pkg/hydrator/nexposeconfig.go (hunk 1 above)
Comment (LorenBoren) [reply pkg/hydrator/nexposeconfig.go]: ✔️ I'll create another issue for that.

File: docker-compose.yaml (hunk 2 above)
Comment (bkrebsbach) [reply docker-compose.yaml]: Suggestion: I'd highly recommend documenting all the environment variables required in the README

Comment (LorenBoren) [reply docker-compose.yaml]: Will do. Thanks!

**Pull request merged**